counterpartylib/lib/blocks.py

```diff
@@ -179,10 +182,11 @@ def parse_block(db, block_index, block_time, previous_ledger_hash=None,
     cursor.close()
 
     # Calculate consensus hashes.
-    new_txlist_hash = check.consensus_hash(db, 'txlist_hash', previous_txlist_hash, txlist)
-    new_ledger_hash = check.consensus_hash(db, 'ledger_hash', previous_ledger_hash, util.BLOCK_LEDGER)
+    new_txlist_hash, found_txlist_hash = check.consensus_hash(db, 'txlist_hash', previous_txlist_hash, txlist)
+    new_ledger_hash, found_ledger_hash = check.consensus_hash(db, 'ledger_hash', previous_ledger_hash, util.BLOCK_LEDGER)
+    new_messages_hash, found_messages_hash = check.consensus_hash(db, 'messages_hash', previous_messages_hash, database.BLOCK_MESSAGES)
 
-    return new_ledger_hash, new_txlist_hash
+    return new_ledger_hash, new_txlist_hash, new_messages_hash, found_messages_hash
```

Comment:
I thought about replacing `ledger_hash` with it. It's an option, but I think `ledger_hash` has value, as it's more strict (we throw exceptions), we do checkpoints on it. `txlist_hash` I'd probably leave as is, because IIRC, it's more of a hash on the input data (transaction list coming in off of the parsed blockchain).

_If_ we enforce `message_hash` with checkpoints and throwing exceptions, like we do with `ledger_hash`, then it might make sense to replace `ledger_hash` with it... BUT, keeping the two could make sense as well, as if `messages_hash` is off and `ledger_hash` is not, you can immediately tell that the database data is different, in a way that does not impact balances. With just `messages_hash`, you don't get that same specificity.